### ISSUE TITLE
[FEATURE] New rule proposal: `no-invalid-entity`

### ISSUE THREAD

**Issue by yeonjuan** [yeonjuan/html-eslint#351]
**Is your feature request related to a problem? Please describe.**
I'd like to propose a new rule called `no-invalid-entity`.
HTML allows the use of named and numeric character references (entities) such as `&nbsp;`, `&#160;`, or `&#xA0;`. However, it's easy to introduce typos like `&nbsb;` or malformed numeric references like &#zzzz;, which are silently ignored by browsers and can lead to rendering issues or confusion.

**correct**
```html
<p>&lt; &gt; &amp; &nbsp; &#160; &#xA0;</p>
```

**incorrect**
```html
<p>&nbsb;</p>        <!-- typo -->
<p>&unknown;</p>     <!-- undefined entity -->
<p>&#zzzz;</p>       <!-- invalid numeric reference -->
```

**Additional context**
https://html.spec.whatwg.org/entities.json

**Comment by perpetual-cosmos** [comment]
Hi! I’m excited about helping implement the no-invalid-entity rule. Though I haven’t built custom lint rules before, I’m eager to learn and collaborate on ensuring HTML entities are properly validated to prevent rendering issues.

Thanks for considering me—looking forward to working together!

**Comment by Danielikenna** [comment]
I suggest a no-invalid-entity rule to catch typos like &nbsb; or &#zzzz;, which browsers ignore, causing issues. See https://html.spec.whatwg.org/entities.js for details. Let me know if you need help

**Comment by yeonjuan** [comment]
@perpetual-cosmos Thanks! Let me know if you have any trouble along the way

**Comment by perpetual-cosmos** [comment]
Hi @yeonjuan ,

Thanks so much for assigning me the task! I'll be starting on it shortly. I really appreciate your support and will definitely reach out if I encounter any difficulties.

**Comment by perpetual-cosmos** [comment]
HI @yeonjuan. I explored the docs/rules/ directory, used no-abstract-roles.md as a reference, and drafted a no-invalid-entity.md file for documentation. I also reviewed the entities.json file from the HTML spec to understand valid entities.

Before I start coding, I’d greatly appreciate your guidance on a couple of points:

Should I add the rule implementation in the packages/ directory, perhaps in a rules/ subdirectory like packages/eslint-plugin-html-eslint/rules/?
For validating entities, would you prefer including a local copy of entities.json in the project, or is there a better approach, like fetching it dynamically?
Thank you so much for your time and for maintaining this amazing project! I’m looking forward to your advice to ensure I’m on the right track.

**Comment by yeonjuan** [comment]
Hi @perpetual-cosmos 
> Should I add the rule implementation in the packages/ directory, perhaps in a rules/ subdirectory like packages/eslint-plugin-html-eslint/rules/?

Yes, there is a script to quickly add a new rule: `yarn run new-rule no-invalid-entity` command will create the necessary files.

>For validating entities, would you prefer including a local copy of entities.json in the project, or is there a better approach, like fetching it dynamically?

I think it's better to place it locally rather than relying on dynamic patching.

**Comment by perpetual-cosmos** [comment]
Hi @yeonjuan ,

Thank you so much for your response! I really appreciate the guidance. 
`I’ll use the **yarn run new-rule no-invalid-entity scrip**t to set up the rule files in packages/eslint-plugin-html-eslint/rules/,` 
and I also believe that `including a local copy of entities.json in the project is a more maintainable and reliable approach`, as relying on external fetch calls increases dependency on external sources and can affect overall reliability..

I’m excited to start implementing the rule and will update this thread with my progress. Thanks again for your support!

**Comment by perpetual-cosmos** [comment]
hey @yeonjuan 
I have successfully implemented the no-invalid-entity rule, thoroughly tested it, and submitted a pull request. 

While testing, **_I encountered an E2E test failure in tests/e2e/index.test.js_** (e2e/robertakarobin), which is unrelated to my changes. I’ve noted this issue in the PR and sought guidance on how to proceed.

> I’m eagerly looking forward to your feedback on the pull request and am ready to graciously address any corrections or improvements I might have missed.

 Thank you for your support throughout this process—it’s been an incredible experience contributing under your guidence.

**Comment by yeonjuan** [comment]
fixed : https://github.com/yeonjuan/html-eslint/pull/371
